rasterio/rio/info.py

```diff
@@ -15,29 +15,153 @@
+                "'%s' is not a recognized CRS." % retval,
+                param=param, param_hint='crs')
+    return retval
+
+
+def like_handler(ctx, param, value):
+    """Copy a dataset's meta property to the command context for access
+    from other callbacks."""
+    if ctx.obj is None:
+        ctx.obj = {}
+    if value:
+        with rasterio.open(value) as src:
+            metadata = src.meta
+            ctx.obj['like'] = metadata
+            ctx.obj['like']['transform'] = metadata['affine']
+            ctx.obj['like']['tags'] = src.tags()
+
+
+def nodata_handler(ctx, param, value):
+    """Get nodata value from a template file or command line."""
+    retval = from_like_context(ctx, param, value)
+    if retval is None and value is not None:
+        try:
+            retval = float(value)
+        except:
+            raise click.BadParameter(
+                "%s is not a number." % repr(value),
+                param=param, param_hint='nodata')
+    return retval
+
+
+def tags_handler(ctx, param, value):
+    """Get tags from a template file or command line."""
+    retval = from_like_context(ctx, param, value)
+    if retval is None and value:
+        try:
+            retval = dict(p.split('=') for p in value)
+        except:
+            raise click.BadParameter(
+                "'%s' contains a malformed tag." % value,
+                param=param, param_hint='transform')
+    return retval
+
+
+def transform_handler(ctx, param, value):
+    """Get transform value from a template file or command line."""
+    retval = from_like_context(ctx, param, value)
+    if retval is None and value:
+        try:
+            value = json.loads(value)
+        except ValueError:
+            pass
+        try:
+            retval = guard_transform(value)
+        except:
+            raise click.BadParameter(
+                "'%s' is not recognized as an Affine or GDAL "
+                "geotransform array." % value,
+                param=param, param_hint='transform')
+    return retval
+
+
+# The edit-info command.
+
 @click.command('edit-info', short_help="Edit dataset metadata.")
 @options.file_in_arg
-@click.option('--nodata', type=float, default=None,
+@click.option('--nodata', callback=nodata_handler, default=None,
               help="New nodata value")
-@click.option('--crs', help="New coordinate reference system")
-@click.option('--transform', help="New affine transform matrix")
-@click.option('--tag', 'tags', multiple=True, metavar='KEY=VAL',
-              help="New tag.")
+@click.option('--crs', callback=crs_handler, default=None,
+              help="New coordinate reference system")
+@click.option('--transform', callback=transform_handler,
+              help="New affine transform matrix")
+@click.option('--tag', 'tags', callback=tags_handler, multiple=True,
+              metavar='KEY=VAL', help="New tag.")
+@click.option('--all', 'allmd', callback=all_handler, flag_value='like',
```

Comment:
@sgillies even though this default is set to ```False``` the effect still seems to be as if it were ```True```.  If I call 

```rio edit-info example.tif --like template.tif```

it produces the same result as 

``` rio edit-info example.tif --like template.tif --all```